### DISCUSSION TITLE
Raspberry Pi ERR: path must end with a slash (cam)

### DISCUSSION THREAD

**Discussion by mlittle408** [bluenviron/mediamtx#675] [Questions]
## Which version are you using?

v0.17.6

## Which operating system are you using?

OS

- [x] Linux
- [ ] Windows
- [ ] macOS

Architecture

- [ ] amd64
- [ ] arm64
- [ ] arm7
- [x] arm6

## Describe the issue

Unable to connect to stream from vlc log file shows 
2021/10/15 13:16:40 I [0/0] [RTSP] [conn 192.168.1.156:64251] ERR: no one is publishing to path 'cam'

## Describe how to replicate the issue

Raspberry pi zero running Buster with Pi camera module 2

I installed Gstreamer according to the wiki with:
sudo apt install -y gstreamer1.0-tools gstreamer1.0-rtsp

I also installed gst-rpicamsrc according to the documentation:
apt-get install autoconf automake libtool pkg-config libgstreamer1.0-dev \
 libgstreamer-plugins-base1.0-dev libraspberrypi-dev

and 

./autogen.sh --prefix=/usr --libdir=/usr/lib/arm-linux-gnueabihf/
make
sudo make install

I also ran a test gst-inspect-1.0 rpicamsrc and did not get any errors. The output is attached as gst-inspect-results.txt

I also ran gst-launch-1.0 rpicamsrc bitrate=1000000 ! filesink location=test.h264. 

Test.h264 was created and viewing the file in vlc confirmed that the camera is working.
  
Running this from the command line also works:
gst-launch-1.0 rpicamsrc preview=false bitrate=2000000 keyframe-interval=50 ! video/x-h264,width=640,height=480,framerate=25/1 ! filesink location=test.h264

## Did you attach a network dump?

no

[config-yml.txt](https://github.com/aler9/rtsp-simple-server/files/7353445/config-yml.txt)
[gst-inspect-results.txt](https://github.com/aler9/rtsp-simple-server/files/7353448/gst-inspect-results.txt)
[log-output.txt](https://github.com/aler9/rtsp-simple-server/files/7353450/log-output.txt)

**Comment by aler9** [comment]
In the log there are no traces of gst-launch-1.0 running - please make sure that

* you're not using the Docker version (which doesn't have access to any executable in the host, including gst-launch-1.0)
* gst-launch-1.0 is accessible from rtsp-simple-server.

**Comment by mlittle408** [comment]
I'm not using the Docker version and I followed the wiki instructions. I've gone back through and tried again but am running into the same problem.

**Comment by mlittle408** [comment]
I did some testing and found that this works:

  runOnInit: gst-launch-1.0 rpicamsrc preview=false bitrate=2000000 keyframe-interval=50 ! video/x-h264,width=1920,height=1080,framerate=25/1 ! rtspclientsink location=rtsp://localhost:$RTSP_PORT/$RTSP_PATH

While this doesn't:
   runOnInit: gst-launch-1.0 rpicamsrc preview=false bitrate=2000000 keyframe-interval=50 ! video/x-h264,width=1920,height=1080,framerate=25/1 ! h264parse ! rtspclientsink location=rtsp://localhost:$RTSP_PORT/$RTSP_PATH

The problem is with ! h264parse !

A little more digging and I learned that h264parse is part of GStreamer Bad Plug-ins

Once I installed that, things worked fine. 

sudo apt-get install gstreamer1.0-plugins-bad

**Comment by aler9** [comment]
h264parse is necessary, since it extract two fundamental parameters from the stream (SPS and PPS) that are sent to the server through the RTSP handshake, then sent from the server to the readers through the RTSP handshake and are needed to correctly decode H264.

**Comment by aler9** [comment]
`sudo apt-get install gstreamer1.0-plugins-bad` was added to the README

[a comment by an automated account is not shown]
